src/coreComponents/constitutive/fluid/multifluid/blackOil/BlackOilFluid.hpp

```diff
@@ -529,8 +529,8 @@ BlackOilFluid::KernelWrapper::
     phaseFraction.value[ipWater] = zw;
 
     // oil
-    phaseCompFraction.value[ipOil][icOil] = zo;
-    phaseCompFraction.value[ipOil][icGas] = zg;
+    phaseCompFraction.value[ipOil][icOil] = zo / ( zo + zg );
+    phaseCompFraction.value[ipOil][icGas] = zg / ( zo + zg );
```

Comment:
I would have written this as for example `zo / (1 - zw)` which should have no impact here but would have an impact on the derivatives. Have you looked at the Jacobian in this alternative formulation. I find these composition derivatives a bit confusing when honouring that they sum to 1.

Reply:
just checked - Jacobian is the same, number of linear iterations is the same, I will update with your version